dbengine/elasticsearch_engine_test.go

```diff
@@ -0,0 +1,143 @@
+package dbengine
+
+import (
+	composeapi "github.com/compose/gocomposeapi"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	httpmock "gopkg.in/jarcoal/httpmock.v1"
+)
+
+var _ = Describe("ElasticSearch", func() {
+	var (
+		credentials Credentials
+		dbEngine    *ElasticSearchEngine
+		deployment  composeapi.Deployment
+	)
+
+	BeforeEach(func() {
+		credentials = Credentials{
+			URI: getEnvOrDefault("ES_URL", "http://localhost:9200"),
+		}
+
+		deployment = composeapi.Deployment{
+			Connection: composeapi.ConnectionStrings{
+				Direct: []string{credentials.URI},
+			},
+		}
+	})
+
+	Context("Connection Testing", func() {
+		BeforeEach(func() {
+			dbEngine = &ElasticSearchEngine{credentials: &credentials}
+		})
+
+		It("should testConnection() successfully", func() {
+			err := dbEngine.testConnection()
+
+			Expect(err).ShouldNot(HaveOccurred())
+		})
+
+		It("should fail to testConnection() due to faulty url", func() {
+			dbEngine.credentials.URI = "http://localhost:8080"
+			err := dbEngine.testConnection()
+
+			Expect(err).Should(HaveOccurred())
```

Comment:
I'm more keen on that, based on the fact that we do similar thing for RDS broker. And I know, we shouldn't blindly duplicate the two, but it sounds fairly logical to do it this way.